### DISCUSSION TITLE
Robot Setup Instructions

### DISCUSSION THREAD

**Discussion by ted-miller** [Yaskawa-Global/YMConnect#4] [General]
@aseamans, I just realized that the robot setup instructions are not included in the documentation. Lets create a page for that and provide a link in this discussion.

**Comment by RubyAI132** [comment]
Hello 
I get a failure when trying to run the Hello world (quick start ) project 
I didn't made any setting adjustment to the controller. Do I need do sett something in the controller?
I have a YRC1000 micro controller and I'm using visual studio 2022 to compile and run/ debug the solution
I changed the ip address parameter in OpenConnection to the ip of the controlee and compiled the sample code successfully 
I made sure I have ping to the controller ip and started debug .

The issue I see is that there is no response  from the controller 
This is the message I see on the console 
'Code (196614): No response from controller after 3 retries.'

The code line that fails is this 
code line 1378-1380
    // proxy allocated, drain it
    for (auto _Pnext = _STD exchange(_Myproxy->_Myfirstiter, nullptr); _Pnext; _Pnext = _Pnext->_Mynextiter) {
        _Pnext->_Myproxy = nullptr;

**Reply by aseamans** [reply]
Hello
Today, we went ahead and created a page for instructions to [set up the robot controller for YMConnect.](https://developer.motoman.com/en/YMConnect/RobotControllerSetup)

Thanks!

**Reply by ted-miller** [reply]
To clarify, the message `Code (196614): No response from controller after 3 retries.` indicates that the PC was unable to connect to the robot controller. This is most likely due to a misconfigured parameter. The link from aseamans provides the required settings.

When `OpenConnection` fails to connect to the robot controller, it returns a null pointer. The failure on `code line 1378-1380` is most likely due to trying to use the null pointer.

**Reply by RubyAI132** [reply]
Hello again  
I followed the controller setup guide as well as the Motoman ethernet function manual
It turned out my controller factory setting had the High Speed Ethernet function enabled .
But still I can't establish connection for simple hello world 

I did packet capture  with Wireshark on my PC (windows10) network interface  while trying to run the quick start code and I observed no! packet were sent out from the PC with the controller address .

I made sure firewalls are disabled and that there is ping to the controller but no success to run Hello World
Do I need to look / try something else 

Thank you for any help

**Reply by aseamans** [reply]
Hello,
Are you seeing any network packets at all when you start your program? It's interesting if there are no packets that even try to get sent out from YMConnect. 

-Austen

**Comment by RubyAI132** [comment]
I see the ping  going out

Sent from Mail for Windows

**Reply by aseamans** [reply]
By ping do you mean the initial packets sent from the "Hello YMConnect" sample code?

**Reply by aseamans** [reply]
Also, if you can provide a cmos.bin of your controller, the packet capture that you are seeing and the IP addresses for your pc and the controller, I can try to replicate your issue.

**Comment by RubyAI132** [comment]
The ping has nothing to do with Hello YMConnect,  it’s a sanity test  I do to the setup.

When I run just YMConnect there is nothing going out.
I also use the controller High speed ethernet error log utility  and don’t see any activity

Tank you 

Sent from Mail for Windows

**Comment by RubyAI132** [comment]
Hello see attached packet  dump 

 The controller is 192.168.100.1
And PC is 192.168.100.77

Sent from Mail for Windows

**Reply by ted-miller** [reply]
The message did not include the attachment. Could you please provide the attachment on the Github page. Also, a copy of the CMOS.BIN from your robot controller would be helpful to try and reproduce.

**Comment by RubyAI132** [comment]
Hello

The attached archive  has  the COMS.BIN and the packet dump
 I send ping prior to starting the YMC Hello World 
The dump period includes both the ping and the YMC Hello World until failing 

The controller ip is 192.168.100.1
And PC is 192.168.100.77

Thank you  for helping 
Sent from Mail for Windows

**Reply by ted-miller** [reply]
The attachments from your email message are not coming through. Please login to the Github web interface and post them to the discussion board here: https://github.com/Yaskawa-Global/YMConnect/discussions/4

**Reply by RubyAI132** [reply]
[YMCONNECT.zip](https://github.com/Yaskawa-Global/YMConnect/files/14669255/YMCONNECT.zip)


hopefully now

The attached ZIP has  the COMS.BIN and the packet dump
I send ping prior to starting the YMC Hello World 
The dump period includes both the ping and the YMC Hello World until failing 

The controller ip is 192.168.100.1
And PC is 192.168.100.77

**Comment by ted-miller** [comment]
This discussion board is a bit off topic. I've opened a new one here: https://github.com/Yaskawa-Global/YMConnect/discussions/7
